3drepo/3drepo.io
# Issue 3397 - Edit Profile, Link, Unlink endpoints

This fixes #3397

#### Description
1) Get Profile endpoint now returns isSso property
2) Edit Profile endpoint does not allow SSO user to edit email, password, firstname and lastname
3) Link - Non SSO users can now link up to SSO
4) Unlink - SSO users can now become normal users

#### Test cases
1) Make a call to Get Profile endpoint. If a user is SSO an isSso property will be returned
2) Try editing one of the 4 fields from an SSO user account. An error should be thrown
3) Call link endpoint from a normal users account. After Ms authentication the user should become SSO
4) Call unlink endpoint from an SSO user providing a password and they will become normal user
